cabal-install/src/Distribution/Client/CmdInstall.hs

```diff
@@ -987,41 +1076,85 @@ disableTestsBenchsByDefault configFlags =
     , configBenchmarks = Flag False <> configBenchmarks configFlags
     }
 
+symlink :: OverwritePolicy -> InstallExe -> UnitId -> UnqualComponentName -> Symlink
+symlink
+  overwritePolicy
+  InstallExe{installDir, mkSourceBinDir, mkExeName, mkFinalExeName}
+  unit
+  exe =
+    Symlink
+      overwritePolicy
+      installDir
+      (mkSourceBinDir unit)
+      (mkExeName exe)
+      (mkFinalExeName exe)
+
+errorMessage :: Pretty a => OverwritePolicy -> InstallMethod -> FilePath -> a -> String
+errorMessage overwritePolicy installMethod installdir exe = case overwritePolicy of
+  NeverOverwrite ->
+    "Path '"
+      <> (installdir </> prettyShow exe)
+      <> "' already exists. "
+      <> "Use --overwrite-policy=always to overwrite."
+  -- This shouldn't even be possible, but we keep it in case
+  -- symlinking/copying logic changes
+  _ ->
+    case installMethod of
+      InstallMethodSymlink -> "Symlinking"
+      InstallMethodCopy ->
+        "Copying" <> " '" <> prettyShow exe <> "' failed."
+
+-- | Check if we can Symlink/copy every exe from a package from the store to a given location
```

Comment:
It's a little sad to see a haddock just sayong "check X" on a function returning `IO ()`. I'd rather have separate pure checking function or a longer haddock explaining what happens if check succeeds/fails.

Reply:
I improved the message from "check" to "try" and "stop".